src/main/java/cn/nukkit/block/BlockPistonBase.java

```diff
@@ -228,7 +228,7 @@ private boolean doMove(boolean extending) {
                                 .putInt("id", newBlock.getId()) //only for nukkit purpose
                                 .putInt("meta", newBlock.getDamage()) //only for nukkit purpose
                                 .putShort("val", newBlock.getDamage())
-                                .putString("name", GlobalBlockPalette.getName(newBlock.getId()))
+                                .putString("name", newBlock.getName())
```

Comment:
The client expect it to be a valid string-id like `minecraft:cobblestone`, that information is only accurately available at the `runtime_block_states.data` files which are (currently) interpreted by the `GlobalBlockPalette`. If an empty string got into that spot, than the issue is with the names registered in the `GlobalBlockPalette` class